### PR TITLE
Fix row number mismatch

### LINKED ISSUE #22305: [Bug]: Row number is not accurate, which block flush
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. when update growing segment current row number, it should be aligned with flushed segment row number, but not use current row count in memory
2. when datacoord restart, rebuild current row for growing segment may not be accurate.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by xiaofan-luan**
fix #22305
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

File: internal/datacoord/server.go [hunk 1]
```diff
@@ -994,7 +1008,16 @@ func (s *Server) reCollectSegmentStats(ctx context.Context) {
 	nodes := s.sessionManager.getLiveNodeIDs()
 	log.Info("re-collecting segment stats from DataNodes",
 		zap.Int64s("DataNode IDs", nodes))
-	for _, node := range nodes {
-		s.cluster.ReCollectSegmentStats(ctx, node)
+
+	reColllectFunc := func() error {
```
Comment (congqixia) [inline internal/datacoord/server.go:1012]: NIT: typo

File: internal/datacoord/server.go [hunk 2]
```diff
@@ -994,7 +1008,16 @@ func (s *Server) reCollectSegmentStats(ctx context.Context) {
...
+		err := s.cluster.ReCollectSegmentStats(ctx)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := retry.Do(ctx, reColllectFunc, retry.Attempts(20)); err != nil {
```
Comment (congqixia) [inline internal/datacoord/server.go:1020]: `Attempts(20)` will cost about 50 seconds. May caused unwanted panic when cluster restarts and datanodes' sessions remains in etcd (worst case 60 seconds)

**Review by congqixia — APPROVED** [review]
overall lgtm
please check comments below
/lgtm

File: internal/datacoord/server.go (hunk 2 above)
Comment (xiaofan-luan) [reply internal/datacoord/server.go:1020]: set the max retry time to 5s to avoid the corner case.
We will change etcd lease timeout to 20s

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**